app/controllers/application_controller.rb

```diff
@@ -18,7 +18,8 @@ def pundit_user
 
   def auth_user
     unless user_signed_in?
-      redirect_to root_path, notice: 'You must sign in first!'
+      $baseURL = request.fullpath
+      redirect_post(user_saml_omniauth_authorize_path, options: {authenticity_token: :auto})
     end
```

Comment:
I got the error: undefined method `redirect_post' for line 22 in the applicaton_controller

Reply:
A new gem is added. Please run delete Gemfile.lock file, run bundle install and then try again.